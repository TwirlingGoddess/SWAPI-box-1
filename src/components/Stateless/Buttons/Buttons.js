import './Buttons.css';
import React, { Component } from 'react';
import { NavLink, Route } from 'react-router-dom'
import PropTypes from 'prop-types';



const Buttons = ({makeApiCall,favoritesLength,displayFavorites}) => {

  return (
    <div>
      <button><NavLink to='/people' className='people nav' onClick={() => makeApiCall('people')}>People</NavLink></button>
      <button><NavLink to='/planets' className='planets nav' onClick={() => makeApiCall('planets')}>Planets</NavLink></button>
      <button><NavLink to='/vehicles' className='vehicles nav' onClick={() => makeApiCall('vehicles')}>Vehicles</NavLink></button>
      <button><NavLink to='/favorites' className='favorite nav' onClick={() => displayFavorites()}>Favorite:{favoritesLength}</NavLink></button>
    </div>
  )
}

// Buttons.propTypes = {
//   makeApiCall: PropTypes.func.isRequiered,
//   favoritesLength: PropTypes.number.isRequired,
//   displayFavorites: PropTypes.func.isRequired
// };


export default Buttons;