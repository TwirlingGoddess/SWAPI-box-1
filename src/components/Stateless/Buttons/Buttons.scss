$black-color :  black;
$main-font : 'Nanum Gothic Coding', monospace;

button {
  background-color: transparent;
  background-image: url('../../../assets/star-wars-2369317_640.png');
  background-repeat: space;
  background-size: 21vh;
  border: none;
  color: white;
  cursor: pointer;
  font-family: 'Nanum Gothic Coding', monospace;  
  font-size: 4vh;
  height: 12vh;
  margin: 1vh;
  padding: 3vh;
  width: 25vh;
}

a {
  color: white;
  text-decoration: none
}
