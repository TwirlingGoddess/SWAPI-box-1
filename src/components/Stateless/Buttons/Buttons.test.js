import React from 'react';
import { shallow, mount } from 'enzyme';
import Buttons from './Buttons';


describe('Buttons', () => {
  let buttons
  let apiCall
  let displayFavorites

  beforeEach(() => {
    apiCall = jest.fn()
    displayFavorites = jest.fn()
    buttons = shallow(<Buttons apiCall={apiCall} displayFavorites={displayFavorites}/>);
  })
  it('matches snapshot', () => {
    expect(buttons).toMatchSnapshot()
  })

  it('should call apiCall on click', () => {
    expect(apiCall).toHaveBeenCalledTimes(0);
    buttons.find('.people').simulate('click')
    expect(apiCall).toHaveBeenCalledTimes(1);
    
  })

  it('should call displayFavorites on click', () => {
    expect(displayFavorites).toHaveBeenCalledTimes(0);
    buttons.find('.favorite').simulate('click')
    expect(displayFavorites).toHaveBeenCalledTimes(1);
  })
})