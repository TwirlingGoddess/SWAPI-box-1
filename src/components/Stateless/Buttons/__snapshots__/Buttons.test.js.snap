// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Buttons />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="people nav"
                    onClick={[Function]}
                    to="/people"
          >
                    People
          </NavLink>
</button>,
        <button>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="planets nav"
                    onClick={[Function]}
                    to="/planets"
          >
                    Planets
          </NavLink>
</button>,
        <button>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="vehicles nav"
                    onClick={[Function]}
                    to="/vehicles"
          >
                    Vehicles
          </NavLink>
</button>,
        <button>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="favorite nav"
                    onClick={[Function]}
                    to="/favorites"
          >
                    Favorite:
          </NavLink>
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="people nav"
            onClick={[Function]}
            to="/people"
>
            People
</NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "People",
            "className": "people nav",
            "onClick": [Function],
            "to": "/people",
          },
          "ref": null,
          "rendered": "People",
          "type": [Function],
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="planets nav"
            onClick={[Function]}
            to="/planets"
>
            Planets
</NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Planets",
            "className": "planets nav",
            "onClick": [Function],
            "to": "/planets",
          },
          "ref": null,
          "rendered": "Planets",
          "type": [Function],
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="vehicles nav"
            onClick={[Function]}
            to="/vehicles"
>
            Vehicles
</NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Vehicles",
            "className": "vehicles nav",
            "onClick": [Function],
            "to": "/vehicles",
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": [Function],
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="favorite nav"
            onClick={[Function]}
            to="/favorites"
>
            Favorite:
</NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": Array [
              "Favorite:",
              undefined,
            ],
            "className": "favorite nav",
            "onClick": [Function],
            "to": "/favorites",
          },
          "ref": null,
          "rendered": Array [
            "Favorite:",
            undefined,
          ],
          "type": [Function],
        },
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="people nav"
                        onClick={[Function]}
                        to="/people"
            >
                        People
            </NavLink>
</button>,
          <button>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="planets nav"
                        onClick={[Function]}
                        to="/planets"
            >
                        Planets
            </NavLink>
</button>,
          <button>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="vehicles nav"
                        onClick={[Function]}
                        to="/vehicles"
            >
                        Vehicles
            </NavLink>
</button>,
          <button>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="favorite nav"
                        onClick={[Function]}
                        to="/favorites"
            >
                        Favorite:
            </NavLink>
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="people nav"
              onClick={[Function]}
              to="/people"
>
              People
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "People",
              "className": "people nav",
              "onClick": [Function],
              "to": "/people",
            },
            "ref": null,
            "rendered": "People",
            "type": [Function],
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="planets nav"
              onClick={[Function]}
              to="/planets"
>
              Planets
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Planets",
              "className": "planets nav",
              "onClick": [Function],
              "to": "/planets",
            },
            "ref": null,
            "rendered": "Planets",
            "type": [Function],
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="vehicles nav"
              onClick={[Function]}
              to="/vehicles"
>
              Vehicles
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Vehicles",
              "className": "vehicles nav",
              "onClick": [Function],
              "to": "/vehicles",
            },
            "ref": null,
            "rendered": "Vehicles",
            "type": [Function],
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="favorite nav"
              onClick={[Function]}
              to="/favorites"
>
              Favorite:
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": Array [
                "Favorite:",
                undefined,
              ],
              "className": "favorite nav",
              "onClick": [Function],
              "to": "/favorites",
            },
            "ref": null,
            "rendered": Array [
              "Favorite:",
              undefined,
            ],
            "type": [Function],
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
